### PR TITLE
added PRAGMA statement

### PR DESCRIPTION
**Pull request by mpf82**
SQLITE: https://www.sqlite.org/pragma.html

DuckDB: https://duckdb.org/docs/sql/pragmas.html

PL/SQL: https://juliandontcheff.wordpress.com/2023/01/03/the-106-oracle-pl-sql-pragmas/

### CODE REVIEW

File: sqlglot/tokens.py [hunk 1]
```diff
@@ -252,6 +252,7 @@ class TokenType(AutoName):
     PERCENT = auto()
     PIVOT = auto()
     PLACEHOLDER = auto()
+    PRAGMA = auto()
```
Comment (GeorgeSittas) [inline sqlglot/tokens.py:255]: Let's include this in `ID_VAR_TOKENS` as well.

File: sqlglot/parser.py [hunk 2]
```diff
@@ -3880,6 +3881,25 @@ def _parse_merge(self) -> exp.Expression:
             expressions=whens,
         )
 
+    def _parse_pragma(self) -> t.Optional[exp.Expression]:
+        this = self._parse_primary() or self._parse_id_var()
+        schema = None
+        value = None
+        func = None
+
+        while self._match(TokenType.DOT):
```
Comment (GeorgeSittas) [inline sqlglot/parser.py:3890]: I think this can be converted to an if statement instead.

File: sqlglot/parser.py [hunk 3]
```diff
@@ -3880,6 +3881,25 @@ def _parse_merge(self) -> exp.Expression:
...
+            schema = this
+            this = self._parse_primary() or self._parse_id_var()
+
+        if self._match(TokenType.EQ):
+            self._match(TokenType.EQ)
```
Comment (GeorgeSittas) [inline sqlglot/parser.py:3895]: We've already consumed the EQ token here, so this seems redundant unless there's a syntax form I missed that expects `==` as a token.

File: sqlglot/parser.py [hunk 4]
```diff
@@ -3880,6 +3881,25 @@ def _parse_merge(self) -> exp.Expression:
...
+        this = self._parse_primary() or self._parse_id_var()
+        schema = None
+        value = None
+        func = None
+
+        while self._match(TokenType.DOT):
+            schema = this
+            this = self._parse_primary() or self._parse_id_var()
+
+        if self._match(TokenType.EQ):
```
Comment (GeorgeSittas) [inline sqlglot/parser.py:3894]: I think we can dry this out to

```python
func = self._match(TokenType.L_PAREN)
self._match(TokenType.EQ)
value = self._parse_var_or_string() or self._parse_number() or self._parse_id_var()
self._match(TokenType.R_Paren)
```

File: sqlglot/generator.py [hunk 5]
```diff
@@ -1191,6 +1191,17 @@ def set_sql(self, expression: exp.Set) -> str:
         )
         return f"SET{expressions}"
 
+    def pragma_sql(self, expression: exp.Pragma) -> str:
+        this = self.sql(expression, "this")
+        schema = self.sql(expression, "schema")
+        schema = f"{schema}." if schema else ""
+        value = self.sql(expression, "value")
+        func = expression.args.get("func")
+        if func:
```
Comment (GeorgeSittas) [inline sqlglot/generator.py:1200]:
```suggestion
        if expression.args.get("func"):
```

**Review by GeorgeSittas — COMMENTED** [review]
Looks good, just a few comments

File: sqlglot/parser.py (hunk 3 above)
Comment (mpf82) [reply sqlglot/parser.py:3895]: yeah, that's a leftover from a previous attempt where I checked the token w/o advancing

File: sqlglot/parser.py (hunk 4 above)
Comment (mpf82) [reply sqlglot/parser.py:3894]: Changed it the way you suggested, but personally, I think the readability suffers.

**Review by GeorgeSittas — APPROVED** [review]

File: sqlglot/parser.py (hunk 4 above)
Comment (GeorgeSittas) [reply sqlglot/parser.py]: Thanks for making the changes, happy to revert it later on if Toby agrees about the readability issue. It's a common pattern to dry out things like this in the codebase, so I'm probably biased towards this style more.

File: sqlglot/parser.py [hunk 6]
```diff
@@ -3880,6 +3882,22 @@ def _parse_merge(self) -> exp.Expression:
             expressions=whens,
         )
 
+    def _parse_pragma(self) -> t.Optional[exp.Expression]:
+        this = self._parse_primary() or self._parse_id_var()
+        schema = None
+        value = None
```
Comment (tobymao) [inline sqlglot/parser.py:3888]:
```suggestion
```

**Comment by tobymao** [comment]
@mpf82 i've greatly simplified pragma with e8cff1cb

**Pull request merged**